File: HeartShop-Spring/HeartShop/src/main/java/com/HeartShop/entity/Wishlist.java
package com.HeartShop.entity;

import lombok.Data;
import java.math.BigDecimal;
import java.util.Date;

@Data
public class Wishlist {
    private Long wishlistId;
    private Long memberId;
    private Long productId;
    private Date createdAt;

    // Joined from product / product_image
    private String productName;
    private String productNameEn;
    private String productCode;
    private BigDecimal price;
    private BigDecimal discountPrice;
    private String productImage;
}

File: heartshop-frontend/services/wishlist.ts
import { apiFetch } from './http/client'

export interface WishlistItem {
  wishlistId: number
  productId: number
  productName: string
  productNameEn: string
  productCode: string
  price: number
  discountPrice: number
  productImage: string
  createdAt: string
}

export const wishlistService = {
  async getWishlist(): Promise<WishlistItem[]> {
    return apiFetch<WishlistItem[]>('/api/wishlist', { method: 'GET' })
  },

  async checkStatus(productId: number): Promise<boolean> {
    const res = await apiFetch<{ wishlisted: boolean }>(
      `/api/wishlist/${productId}/status`,
      { method: 'GET' }
    )
    return res.wishlisted
  },

  async add(productId: number): Promise<void> {
    await apiFetch(`/api/wishlist/${productId}`, { method: 'POST' })
  },

  async remove(productId: number): Promise<void> {
    await apiFetch(`/api/wishlist/${productId}`, { method: 'DELETE' })
  },

  async toggle(productId: number): Promise<boolean> {
    const isWishlisted = await this.checkStatus(productId)
    if (isWishlisted) {
      await this.remove(productId)
      return false
    } else {
      await this.add(productId)
      return true
    }
  }
}
